src/k8s/pkg/k8sd/api/cluster_config.go

```diff
@@ -0,0 +1,212 @@
+		if err != nil {
+			return response.InternalError(fmt.Errorf("failed to reconcile local-storage: %w", err))
+		}
+	}
+
+	if req.Config.Gateway != nil {
+		err := component.ReconcileGatewayComponent(r.Context(), snap, oldConfig.Gateway.Enabled, req.Config.Gateway.Enabled, clusterConfig)
+		if err != nil {
+			return response.InternalError(fmt.Errorf("failed to reconcile gateway: %w", err))
+		}
+	}
+
+	if req.Config.Ingress != nil {
+		err := component.ReconcileIngressComponent(r.Context(), snap, oldConfig.Ingress.Enabled, req.Config.Ingress.Enabled, clusterConfig)
+		if err != nil {
+			return response.InternalError(fmt.Errorf("failed to reconcile ingress: %w", err))
+		}
+	}
+
+	if req.Config.LoadBalancer != nil {
+		err := component.ReconcileLoadBalancerComponent(r.Context(), snap, oldConfig.LoadBalancer.Enabled, req.Config.LoadBalancer.Enabled, clusterConfig)
+		if err != nil {
+			return response.InternalError(fmt.Errorf("failed to reconcile load-balancer: %w", err))
+		}
+	}
+
+	if req.Config.MetricsServer != nil {
+		err := component.ReconcileMetricsServerComponent(r.Context(), snap, oldConfig.MetricsServer.Enabled, req.Config.MetricsServer.Enabled, clusterConfig)
+		if err != nil {
+			return response.InternalError(fmt.Errorf("failed to reconcile metrics-server: %w", err))
+		}
+	}
+
+	return response.SyncResponse(true, &api.UpdateClusterConfigResponse{})
+}
+
+func getClusterConfig(s *state.State, r *http.Request) response.Response {
+	cfg, err := utils.GetClusterConfig(r.Context(), s)
+	if err != nil {
+		return response.InternalError(fmt.Errorf("failed to get cluster config: %w", err))
+	}
+
+	userFacing := api.UserFacingClusterConfig{
+		Network: &api.NetworkConfig{
+			Enabled: vals.Pointer(true),
+		},
+		DNS: &api.DNSConfig{
+			Enabled:             vals.Pointer(true),
+			UpstreamNameservers: cfg.DNS.UpstreamNameservers,
+			ServiceIP:           cfg.Kubelet.ClusterDNS,
+			ClusterDomain:       cfg.Kubelet.ClusterDomain,
+		},
+		Ingress: &api.IngressConfig{
+			Enabled:             vals.Pointer(false),
+			DefaultTLSSecret:    cfg.Ingress.DefaultTLSSecret,
+			EnableProxyProtocol: vals.Pointer(false),
+		},
+		LoadBalancer: &api.LoadBalancerConfig{
+			Enabled:        vals.Pointer(false),
+			L2Enabled:      vals.Pointer(false),
+			L2Interfaces:   cfg.LoadBalancer.L2Interfaces,
+			BGPEnabled:     vals.Pointer(false),
+			BGPLocalASN:    cfg.LoadBalancer.BGPLocalASN,
+			BGPPeerAddress: cfg.LoadBalancer.BGPPeerAddress,
+			BGPPeerASN:     cfg.LoadBalancer.BGPPeerASN,
+			BGPPeerPort:    cfg.LoadBalancer.BGPPeerPort,
+		},
+		LocalStorage: &api.LocalStorageConfig{
+			Enabled:       vals.Pointer(false),
+			LocalPath:     cfg.LocalStorage.LocalPath,
+			ReclaimPolicy: cfg.LocalStorage.ReclaimPolicy,
+			SetDefault:    vals.Pointer(true),
+		},
+		Gateway: &api.GatewayConfig{
+			Enabled: vals.Pointer(false),
+		},
+		MetricsServer: &api.MetricsServerConfig{
+			Enabled: vals.Pointer(false),
```

Comment:
this should be enabled by default, shouldn't it?

Reply:
This is for initializing an empty struct, the default components to be enabled comes from the `BootstrapConfig` as implemented previously. Check `ClusterConfigFromBootstrapConfig` to see how it's adjusted.